experimental/sdk/server-proxies/src/tunnels/TunnelConnection.ts

```diff
@@ -253,37 +253,46 @@ export class TunnelConnection {
       yield `${clientId}: [${client.currentConnectionId}] ended? ${client.stopped}`;
     }
   }
+
+  private getUrl(target: ConnectionTarget, hub: string): { endpoint: URL; reverseProxyEndpoint: URL | undefined } {
+    const HttpTunnelPath = "server/tunnel";
+    let endpoint;
+    if (target.endpoint) {
+      endpoint = target.endpoint;
+    } else {
+      logger.info(`No endpoint from 'target' use original endpoint ${this.endpoint}`);
+      endpoint = this.endpoint;
+    }
+    const uriBuilder = new URL(endpoint);
+    uriBuilder.protocol = uriBuilder.protocol.toLowerCase() === "http:" ? "ws:" : "wss:";
+  
+    uriBuilder.pathname = appendPath(uriBuilder.pathname, HttpTunnelPath);
+    const hubQuery = `hub=${encodeURIComponent(hub)}`;
+    if (!uriBuilder.search) {
+      uriBuilder.search = `?${hubQuery}`;
+    } else {
+      uriBuilder.search = `${uriBuilder.search}&${hubQuery}`;
+    }
+  
+    if (target.target){
+      uriBuilder.search = `${uriBuilder.search}&${encodeURIComponent(target.target)}`;
+    }
+  
+    let reverseProxy: URL | undefined = undefined;
+    if (target.reverseProxyEndpoint) {
+      reverseProxy = new URL(target.reverseProxyEndpoint);
+      reverseProxy.protocol = reverseProxy.protocol.toLowerCase() === "http:" ? "ws:" : "wss:";
+      reverseProxy.pathname = appendPath(reverseProxy.pathname, HttpTunnelPath);
+      reverseProxy.search = uriBuilder.search;
+    }
+  
+    return { endpoint: uriBuilder, reverseProxyEndpoint: reverseProxy };
+  }
 }
 
 function appendPath(pathname: string, append: string): string {
   return pathname.endsWith("/") ? `${pathname}${append}` : `${pathname}/${append}`;
 
 }
 
-function getUrl(target: ConnectionTarget, hub: string): { endpoint: URL; reverseProxyEndpoint: URL | undefined } {
```

Comment:
pass in default endpoint `this.endpoint` is enough?

Reply:
In case one day you will pass another endpoint in message